packages/material-ui/src/LinearProgress/LinearProgress.d.ts

```diff
@@ -7,6 +7,7 @@ export interface LinearProgressProps
   value?: number;
   valueBuffer?: number;
   variant?: 'determinate' | 'indeterminate' | 'buffer' | 'query';
+  styles: {};
```

Comment:
I would do the following:
```ts
export type LinearProgressStyleKeys =
  | 'bar1'
  | 'bar2';
```
and then, `styles: {}` can be changed to
```ts
styles: { [P in LinearProgressStyleKeys]?: React.CSSProperties };
```

For now, regarding the type definition, this can be seen as a temporary fix. In the long run, we should introduce the `styles` api and standardize it just like `XXXClassKey` type. (Probably add `XXXStylesKey` as the third generic type parameter to `StandardProps`? For example, `StandardProps<React.HTMLAttributes<HTMLDivElement>, LinearProgressClassKey, LinearProgressStylesKey>` in this case. @pelotom what do you think?)

Reply:
And also, regarding the failed tests, please make `styles` optional, like this:
```ts
styles?: { [P in LinearProgressStylesKeys]?: React.CSSProperties };
```